### PR TITLE
Firebase Auth and routing after login

### LINKED ISSUE #42: FR18. User Authentication
| FR18. User Authentication |
|-----------------------|
| Goal: FR18 Authenticate user through log in page|
| Stakeholders: All |
| Description: When the user wants to log in, they will first have to verify their username and password. The information will be compared to the firebase database and authenticate the user|
| Origin: Team meeting on 3/23 |
| Version: 1.0 |
| Date: 03/23/2023 |
| Priority: 1|

### LINKED ISSUE #43: FR21. User Authentication Part 2
| FR21. User Authentication Part 2 |
|-----------------------|
| Goal: FR18 Create user registration page that will save the data into firebase to be authenticated|
| Stakeholders: All |
| Description: When the user first opens the app, they will have to create an account that will save the information into firebase to be used to compare when the user wants to log in|
| Origin: Team meeting on 3/23 |
| Version: 1.0 |
| Date: 03/23/2023 |
| Priority: 1|

### PR DESCRIPTION
**Pull request by Ahmed-Alagbari**

### CODE REVIEW

**Review by mxsingh — APPROVED** [review]
Looks good to me! Thanks for working on the routing!

File: lib/cloud/login_page.dart [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+  final _passwordController = TextEditingController();
+
+  void signIn() async{
+   if (_emailController.text.isEmpty || _passwordController.text.isEmpty) {
+  Fluttertoast.showToast(
+        msg: 'Please fill in all fields.');
+  return;
+}
+
+    showDialog(
```
Comment (amuwhao46) [inline lib/cloud/login_page.dart:30]: This shows a new instance of the homepage to each individual "currentUser"?

File: lib/cloud/firestore_cloud_service.dart [hunk 2]
```diff
@@ -1,10 +1,14 @@
...
 import 'package:budgie/cloud/cloud_storage_exceptions.dart';
 import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import '../mainPage.dart';
 import 'cloud_field_constants.dart';
 
 class FirestoreCloudService {
-  final user = FirebaseFirestore.instance.collection('users').doc('test');
+   final user = FirebaseFirestore.instance.collection('users').doc('test');
+   //final User? user=Authenticator().currentUser;
```
Comment (amuwhao46) [inline lib/cloud/firestore_cloud_service.dart:10]: Will this be a part of the final implementation?

File: lib/cloud/login_page.dart [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+                      style: TextStyle(
+                        color: Colors.black,
+                        fontWeight: FontWeight.bold,
+                        fontSize: 18,
+                      ),
+                      )),
+                      ),
+                  )
+                  ),
+                  const SizedBox(height: 25),
```
Comment (amuwhao46) [inline lib/cloud/login_page.dart:184]: SizedBox can be sized dynamically instead of hardcoding a specific height

**Review by amuwhao46 — APPROVED** [review]
Looks really nice. Also I appreciate you reorganizing the files to make it easier to understand the flow of what the current user sees when they are logged in

**Review by JayanthAnandNama — APPROVED** [review]
Looks really good! Great job on the organization like Oke said by the way.

**Review by Rey-Jey — APPROVED** [review]
Code looks good for now, I'll help you out with adding users to Firebase

**Pull request merged**